src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -235,6 +239,32 @@ public void writeLogTo(XMLOutput out) throws IOException {
         @Inject
         private transient GitHubHookRegisterProblemMonitor monitor;
 
+        @Inject
+        private transient SCMTrigger.DescriptorImpl scmTrigger;
+
+        private transient int maximumThreads = Integer.MIN_VALUE;
+
+        public DescriptorImpl() {
+            checkThreadPoolSizeAndUpdateIfNecessary();
+        }
+
+        /**
+         * Update the {@link java.util.concurrent.ExecutorService} instance.
+         */
+        /*package*/
+        synchronized void checkThreadPoolSizeAndUpdateIfNecessary() {
```

Comment:
ookay, i'll merge this. But this is an example of permanent state "will fix that better in future releases". It can't be finished while you've start do things right everywhere